thunder/__init__.py

```diff
@@ -211,6 +211,19 @@ def _get_cache_info():
     return _cache_info_ctx.get()
 
 
+def _get_thunder_module(model):
+    cd = get_compile_data()
+    tm = cd._thunder_module_map.get(id(model))
```

Comment:
purely my taste but
```python
    if (model_hash := id(model) not in cd._thunder_module_map):
        return model
    else:
        tm = cd._thunder_module_map[model_hash]
        if tm._model is not model:
            return model
        return tm
```

Reply:
I should add a comment, but I would view this as a sort of "weak reference", so 217+218 are part of the lookup.